Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -258,6 +271,77 @@ bool Renderer::CheckForHostConfigChanges()
   return true;
 }
 
+bool Renderer::CreateRasterFontResources()
+{
+  // If we don't have a backbuffer, the OSD is not used.
+  if (m_backbuffer_format == AbstractTextureFormat::Undefined)
+    return true;
+
+  // Create texture.
+  TextureConfig texture_config(RasterFont::TEXTURE_WIDTH, RasterFont::TEXTURE_HEIGHT, 1, 1,
+                               AbstractTextureFormat::RGBA8, false);
+  m_raster_font_texture = CreateTexture(texture_config);
+  if (!m_raster_font_texture)
+  {
+    PanicAlert("Failed to create raster font texture.");
+    return false;
+  }
+
+  // Upload texture.
+  std::vector<u8> texture_data = RasterFont::GenerateRGBATexture();
+  m_raster_font_texture->Load(0, RasterFont::TEXTURE_WIDTH, RasterFont::TEXTURE_HEIGHT,
+                              RasterFont::TEXTURE_WIDTH, texture_data.data(), texture_data.size());
+
+  // Create vertex format.
+  PortableVertexDeclaration vdecl = {};
+  vdecl.position = {VAR_FLOAT, 2, offsetof(RasterFont::Vertex, x), true, false};
+  vdecl.texcoords[0] = {VAR_FLOAT, 2, offsetof(RasterFont::Vertex, u), true, false};
+  vdecl.colors[0] = {VAR_UNSIGNED_BYTE, 4, offsetof(RasterFont::Vertex, color), true, false};
+  vdecl.stride = sizeof(RasterFont::Vertex);
+  m_raster_font_vertex_format = g_vertex_manager->CreateNativeVertexFormat(vdecl);
+  _assert_(m_raster_font_vertex_format != nullptr);
+
+  // Compile shaders.
+  std::string vs_code = RasterFont::GenerateVertexShader(g_ActiveConfig.backend_info.api_type);
+  std::string ps_code = RasterFont::GenerateFragmentShader(g_ActiveConfig.backend_info.api_type);
+  auto vs = CreateShaderFromSource(ShaderStage::Vertex, vs_code.c_str(), vs_code.size());
+  auto ps = CreateShaderFromSource(ShaderStage::Pixel, ps_code.c_str(), ps_code.size());
+  if (!vs || !ps)
+  {
+    PanicAlert("Failed to compile raster font shaders.");
+    return false;
+  }
+
+  // Create pipeline.
+  AbstractGraphicsPipelineConfig pconfig = {};
+  pconfig.vertex_format = m_raster_font_vertex_format.get();
+  pconfig.vertex_shader = vs.get();
+  pconfig.pixel_shader = ps.get();
+  pconfig.rasterization_state.hex = RenderState::GetNoCullRasterizationState().hex;
+  pconfig.rasterization_state.primitive = PrimitiveType::Triangles;
+  pconfig.depth_state.hex = RenderState::GetNoDepthTestingDepthStencilState().hex;
+  pconfig.blending_state.hex = RenderState::GetNoBlendingBlendState().hex;
+  pconfig.blending_state.blendenable = true;
```

Comment:
Potentially silly question: are these individual values set necessary with the `RenderState::GetNoBlendingBlendState()` call above?